Mathlib/Data/Set/Subset.lean

```diff
@@ -0,0 +1,260 @@
+  constructor
+  · rintro ⟨_⟩
+    simp_all only [and_self, exists_const]
+  · rintro ⟨⟨_⟩, ⟨_⟩⟩
+    simp_all only [exists_const, and_self]
+
+@[simp]
+lemma coe_compl : ↑(Dᶜ) = A \ ↑D := by
+  ext
+  simp_all only [mem_image, mem_compl_iff, Subtype.exists, exists_and_right, exists_eq_right,
+    mem_diff, not_exists]
+  constructor
+  · rintro ⟨_⟩
+    simp_all only [not_false_eq_true, forall_true_left, and_self]
+  · intro a
+    simp_all only [not_false_eq_true, exists_const]
+
+@[simp]
+lemma coe_diff : (↑(D \ E) : Set α) = ↑D \ ↑E := by
+  ext
+  simp_all only [mem_diff, mem_image, Subtype.exists, exists_and_right, exists_eq_right, not_exists]
+  constructor
+  · rintro ⟨_⟩
+    simp_all only [exists_const, not_false_eq_true, forall_true_left, and_self]
+  · intro
+    simp_all only [not_false_eq_true, and_true]
+
+@[simp]
+lemma coe_sUnion : ↑(⋃₀ T) = ⋃₀ { (B : Set α) | B ∈ T} := by
+  ext x
+  simp_all only [mem_sUnion, mem_setOf_eq, exists_exists_and_eq_and, mem_image, Subtype.exists,
+    exists_and_right, exists_eq_right]
+  constructor
+  · rintro ⟨hxA, D, hDT, hxD⟩
+    exact ⟨D, hDT, hxA, hxD⟩
+  · rintro ⟨D, hDT, hxA, hxD⟩
+    exact ⟨hxA, D, hDT, hxD⟩
+
+@[simp]
+lemma coe_iUnion : ↑(⋃ (B : β), j B) = ⋃ (B : β), (j B : Set α) := image_iUnion
+
+@[simp]
+lemma coe_sInter (hT : T.Nonempty) : (↑(⋂₀ T) : Set α) = ⋂₀ { (↑B : Set α) | B ∈ T } := by
+  ext x
+  cases' hT with L hL
+  constructor
+  · intro h
+    simp_all only [mem_image, mem_sInter, Subtype.exists, exists_and_right, exists_eq_right,
+      mem_setOf_eq, forall_exists_index, and_imp, forall_apply_eq_imp_iff₂]
+    intro a a_1
+    rcases h with ⟨_, _⟩
+    simp_all only [exists_const]
+  · intro h
+    have haux : x ∈ (L : Set α)
+    · simp only [mem_sInter, mem_setOf_eq, forall_exists_index, and_imp, forall_apply_eq_imp_iff₂,
+        mem_image, Subtype.exists, exists_and_right, exists_eq_right] at h
+      specialize h L hL
+      cases' h with y hy
+      use ⟨x, y⟩
+    · simp only [mem_image, mem_sInter, Subtype.exists, exists_and_right, exists_eq_right]
+      rcases haux with ⟨⟨y, hyA⟩, ⟨_, rfl⟩⟩
+      simp_all only [mem_sInter, mem_setOf_eq, forall_exists_index, and_imp,
+        forall_apply_eq_imp_iff₂, mem_image, Subtype.exists, exists_and_right, exists_eq_right,
+        exists_true_left, implies_true, forall_const, exists_const]
+
+@[simp]
+lemma coe_iInter (b : β) : (↑(⋂ (B : β), j B) : Set α) = ⋂ (B : β), (↑(j B) : Set α) := by
+  ext x
+  constructor
+  · intro a
+    simp_all only [mem_image, mem_iInter, Subtype.exists, exists_and_right, exists_eq_right]
+    intro i_1
+    cases a
+    simp_all only [exists_const]
+  · intro h
+    simp only [mem_iInter, mem_image, Subtype.exists, exists_and_right, exists_eq_right] at h
+    cases' h b with hxA hx
+    simp_all only [exists_true_left, mem_image, mem_iInter, Subtype.exists, exists_and_right,
+      exists_eq_right, implies_true, exists_const]
+
+lemma coe_contained : ↑D ⊆ A := by
```

Comment:
`coe_subset`? I'm not sure, but `contained` isn't following the mathlib naming convention.

Reply:
`coe_subset` is already defined. I propose `coe_of_subtype_set_subset`.